src/realm/spec.cpp

```diff
@@ -31,6 +31,13 @@ Spec::~Spec() noexcept
     }
 }
 
+void Spec::init(ref_type ref) noexcept
+{
+    if (!m_top.is_attached() || m_top.get_ref() != ref) {
```

Comment:
Instead of adding a valiable, I have added a comment that should explain what goes on. I could not come up with a name for the variable that would be easier to read than the expression itself.